code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -131,6 +131,8 @@
 		return
 	else
 		SEND_SIGNAL(target, COMSIG_RAT_INTERACT, src)
+	if (DOING_INTERACTION(src, REGALRAT_INTERACTION)) // prevents attacking while performing other interactions (e.g. disposal rummaging)
+		return
```

Comment:
I'm not sure I fully understand what you're suggesting here. Are you saying I should move this early return logic directly into the signal handling for the `COMSIG_RAT_INTERACT` signal? If so, I'll look into that. Admittedly, signal handling in DM is still new to me.

Reply:
Alright, I moved the sleep logic out of the signal, so that this final do_after() check is no longer necessary. Everything should now work as expected. Let me know if there's anything more I need to add/adjust.